src/shared/components/TimePicker/TimePicker.tsx

```diff
@@ -21,58 +24,99 @@ export const TimePicker = <T extends FieldValues>({
   minTime,
   maxTime,
   onCustomChange,
+  defaultTime = DEFAULT_TIME,
   'data-testid': dataTestid,
   placeholder,
   inputSx = {},
-}: TimePickerProps<T>) => (
-  <Controller
-    control={control}
-    name={name}
-    render={({ field: { onChange, value }, fieldState: { error } }) => {
-      const selected = value ? parse(value, DateFormats.Time, new Date()) : value;
-      const handleChange = (date: Date) => {
-        const newTime = dateFnsFormat(date, DateFormats.Time);
-        onCustomChange?.(newTime);
-        onChange(newTime);
-      };
+}: TimePickerProps<T>) => {
+  const [inputValue, setInputValue] = useState<string>('');
+
+  const updateInputValue = (value: string, onChange: (value: string) => void) => {
+    onCustomChange?.(value);
+    setInputValue(value);
+    onChange(value);
+  };
+
+  const handleInputValue = (onChange: (value: string) => void) => {
+    const cleanedInput = cleanInput(inputValue);
+    const validatedInput = validateInput(defaultTime, cleanedInput);
+    const formattedInput = validatedInput.length === 4 ? formatInput(validatedInput) : defaultTime;
+    updateInputValue(formattedInput, onChange);
+  };
+
+  const handleKeyDownEnter = (
+    event: KeyboardEvent<HTMLDivElement>,
+    onChange: (value: string) => void,
+  ) => {
+    if (event.key !== 'Enter') return;
+    handleInputValue(onChange);
+  };
+
+  const handleChange = (
+    date: Date | null,
```

Comment:
It's better to use object for types if there are more than two arguments. 
And, please move types to TimePicker.types.ts. The type for the onChange ((value: string) => void) can be reused.